### PR TITLE
Jv rox 12248 add process id to network protobuf

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Added processed id to network protobuf

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

See above

### CODE REVIEW

**Comment by JoukoVirtanen** [comment]
openshift-newest-qa-e2e-tests fails frequently in other PRs in the same way that it failed here.

**Review by misberner — APPROVED** [review]

File: proto/internalapi/central/network_endpoint_with_process.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+option go_package = "central";
+
+import "google/protobuf/timestamp.proto";
+import "storage/network_flow.proto";
+import "storage/process_indicator.proto";
+import "internalapi/sensor/network_enums.proto";
+
+package central;
+
+message NetworkEndpointWithProcess {
```
Comment (robbycochran) [inline proto/internalapi/central/network_endpoint_with_process.proto:12]: We do not need the listen address based on the MVP requirements. A suggested simplification would be the following:

```
message ProcessListeningOnPortsUpdate {
    repeated storage.ProcessListeningOnPort processes_listening_on_ports = 1;
    google.protobuf.Timestamp time = 2;
}

// keep this definition in storage/
message ProcessListeningOnPort {
    uint32 port = 1;
    storage.L4Protocol protocol = 2;
    storage.ProcessIndicatorUniqueKey process = 3;
    google.protobuf.Timestamp close_timestamp = 4;
}
```

**Review by robbycochran — CHANGES_REQUESTED** [review]

File: proto/storage/process_indicator.proto [hunk 2]
```diff
@@ -47,6 +47,15 @@ message ProcessIndicatorUniqueKey {
     string process_args           = 5;
 }
 
+// This is the processes information which is added to endpoint data
+message ProcessReference {
+    string name           = 1;
+    string agrgs          = 2;
+    string exec_file_path = 3;
+
+    uint32 pid            = 4;
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply proto/storage/process_indicator.proto]: Done

File: proto/storage/process_indicator.proto [hunk 3]
```diff
@@ -47,6 +47,15 @@ message ProcessIndicatorUniqueKey {
     string process_args           = 5;
 }
 
+// This is the processes information which is added to endpoint data
+message ProcessReference {
+    string name           = 1;
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply proto/storage/process_indicator.proto]: Done

File: proto/internalapi/central/network_endpoint_with_process.proto (hunk 1 above)
Comment (JoukoVirtanen) [reply proto/internalapi/central/network_endpoint_with_process.proto]: Done

**Review by dashrews78 — APPROVED** [review]

**Review by Molter73 — APPROVED** [review]
/lgtm

**Review by robbycochran — CHANGES_REQUESTED** [review]

File: proto/internalapi/central/sensor_iservice.proto [hunk 4]
```diff
@@ -14,26 +14,28 @@ import "internalapi/central/baseline_sync.proto";
...
 import "internalapi/central/network_baseline_sync.proto";
 import "internalapi/central/local_scanner.proto";
+import "internalapi/central/process_listening_on_ports_update.proto";
 import "storage/cluster.proto";
 import "storage/image.proto";
 
 package central;
 
 message MsgFromSensor {
-    string hash_key                                                     = 7;
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply proto/internalapi/central/sensor_iservice.proto]: Done

File: proto/internalapi/sensor/network_connection_info.proto [hunk 5]
```diff
@@ -24,11 +25,11 @@ message NetworkConnectionInfo {
 // NetworkConnection tracks connection information about client and server.
 message NetworkConnection {
 
-    SocketFamily socket_family       = 1;
+    SocketFamily socket_family          = 1;
```
[the remark this answers is not in the document]
Comment (JoukoVirtanen) [reply proto/internalapi/sensor/network_connection_info.proto]: Done

**Review by robbycochran — APPROVED** [review]

**Pull request merged**